adapter_windows.go

```diff
@@ -41,6 +41,7 @@ func awaitAsyncOperation(asyncOperation *foundation.IAsyncOperation, genericPara
 	// Wait until the async operation completes.
 	waitChan := make(chan struct{})
 	asyncOperation.SetCompleted(foundation.NewAsyncOperationCompletedHandler(ole.NewGUID(iid), func(instance *foundation.AsyncOperationCompletedHandler, asyncInfo *foundation.IAsyncOperation, asyncStatus foundation.AsyncStatus) {
+		instance.Release()
```

Comment:
I would probably release the handler using `defer`, this way it's clear what we are releasing:
```go
// Wait until the async operation completes.
waitChan := make(chan struct{})
handler := foundation.NewAsyncOperationCompletedHandler(ole.NewGUID(iid), func(instance *foundation.AsyncOperationCompletedHandler, asyncInfo *foundation.IAsyncOperation, asyncStatus foundation.AsyncStatus) {
	status = asyncStatus
	close(waitChan)
})
defer handler.Release()
	
asyncOperation.SetCompleted(handler)
```

Reply:
Thanks. This looks even better